blocks/extensions/stopping.py

```diff
@@ -0,0 +1,35 @@
+from . import FinishAfter
+
+
+class FinishIfNoImprovementAfter(FinishAfter):
+    """Stop after improvements have ceased for a given period.
+
+    Parameters
+    ----------
+    iterations : int
```

Comment:
Agreed, though the question is whether you want to be monitoring for the best at the batch level. In order to then do early stopping at the epoch level, you need to examine every log record for that epoch...